icu4c/source/common/locid.cpp

```diff
@@ -2071,8 +2069,11 @@ Locale::initBaseName(UErrorCode &status) {
                 status = U_MEMORY_ALLOCATION_ERROR;
                 return;
             }
-            copy->fullName.copyFrom(fullName, status);
-            if (U_FAILURE(status)) { return; }
+            copy->fullName = fullName;
+            if (copy->fullName.isEmpty()) {
+                status = U_MEMORY_ALLOCATION_ERROR;
+                return;
+            }
```

Comment:
optional: It would be easier/safer if the FixedString-assign-from-string_view took a UErrorCode, like with CharString, but if you caught every place where it's called, this works.

Reply:
That's just a question of where to put the complexity. All of this is internal so it'll be easy to refactor later, but for now it feels like a good idea to me to have as little complexity in `FixedString` as possible.